src/els_code_action_provider.erl

```diff
@@ -0,0 +1,183 @@
+-module(els_code_action_provider).
+
+-behaviour(els_provider).
+
+-export([ handle_request/2
+        , is_enabled/0
+        , options/0
+        ]).
+
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% els_provider functions
+%%==============================================================================
+
+-spec is_enabled() -> boolean().
+is_enabled() -> true.
+
+%% /**
+%%  * The server provides code actions. The `CodeActionOptions` return type is
+%%  * only valid if the client signals code action literal support via the
+%%  * property `textDocument.codeAction.codeActionLiteralSupport`.
+%%  */
+%% codeActionProvider?: boolean | CodeActionOptions;
+
+-spec options() -> boolean() | map().
+options() ->
+  Capabilities = els_config:get(capabilities),
+  lager:info("code_actions: [Capabilities=~p]", [Capabilities]),
+  true.
+
+-spec handle_request(any(), els_provider:state()) ->
+  {any(), els_provider:state()}.
+handle_request({document_codeaction, Params}, State) ->
+  #{ <<"textDocument">> := #{ <<"uri">> := Uri}
```

Comment:
In maps, we tend to align key/value pairs for readability. As a side note, it would be incredibly cool if one could configure the formatter to do such a thing.

Reply:
cc @michalmuskala